codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -21,11 +21,32 @@ typedef struct {
     int len;
 } MapSet;
 
+/* 用以索引键获取数据 */
+typedef struct {
+    int key[HASH_MAP_DEFAULT_SIZE];
+    int len;
+}IndexMembers;
```

Comment:
`IndexMembers` appears from the code to be part of the `ArrayHashMap` structure, and would be more appropriately placed in an `ArrayHashMap`.

Subsequent function parameters need not be passed.
`IndexMembers *indexMembers,`

Reply:
Yes, but I found an exception while obtaining the ``KeySet`` and ``ValueSet``. Use ``IndexMembers *indexMembers`` is to ensure the right key address.

Reply:
@amlei Could you explain it in detail?